common/src/debug_utils.rs

```diff
@@ -0,0 +1,47 @@
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::Path;
+
+use serde::Serialize;
+
+const DEBUG_FOLDER: &str = "./debug";
+
+/// Serializes a collection of inputs to a pretty-printed JSON format and saves
+/// them to a file.
+///
+/// # Arguments
+///
+/// * `file_name` - The name of the file (including the extension) where the
+///   serialized data will be saved.
+/// * `inputs` - An iterator over a collection of items to be serialized. Each
+///   item in the collection must implement the `Serialize` trait.
+///
+/// # Returns
+///
+/// This function returns a `Result<(), std::io::Error>` indicating the
+/// operation's success or failure.
+pub fn save_inputs_to_disk<I, T>(file_name: String, inputs: I) -> std::io::Result<()>
+where
+    I: IntoIterator<Item = T>,
+    T: Serialize,
+{
+    let input_file_path = Path::new(DEBUG_FOLDER).join(file_name);
+
+    // Ensure the DEBUG_FOLDER exists
+    if fs::metadata(DEBUG_FOLDER).is_err() {
+        fs::create_dir(DEBUG_FOLDER)?;
+    }
+
+    let mut file = File::create(input_file_path)?;
+
+    for input in inputs.into_iter() {
+        // Serialize each input to a pretty JSON string
+        let input_str = serde_json::to_string_pretty(&input)?;
```

Comment:
Is this the best way to do this? Would it be better to do the whole lot at once? With a large collection I'm guessing so